fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/datasource/TableDataSource.java

```diff
@@ -22,10 +22,10 @@
      *
      * @param physicalTable  The physical table of the data source
      */
-    public TableDataSource(PhysicalTable physicalTable) {
+    public TableDataSource(ConcretePhysicalTable physicalTable) {
         super(DefaultDataSourceType.TABLE, Collections.singleton(physicalTable));
 
-        this.name = physicalTable.getFactTableName();
+        this.name = physicalTable.getAvailability().getDataSourceNames().stream().findFirst().get().asName();
```

Comment:
To simplify this, and because `ConcretePhyiscalTable` can only have 1 backing table, could/should we push this logic into `ConcretePhysicalTable` and hide it behind a `getTableName()` method?

And actually, after looking a bit deeper, `ConcretePhyiscalTable` already / still has a `getFactTableName` method that does this logic, so we should use that and rely on the contract / concept of `ConcretePhyiscalTable`.